src/main.rs

```diff
@@ -197,6 +201,8 @@ enum Docs {
 fn main() {
     initialise_logger();
 
+    let mut exit_code = 0;
```

Comment:
Sure.  The thought behind this was that the language server needs to return code 1 if it receives an exit notification before a shutdown notification, which is not an error the end-user needs to be aware of, as it's a consequence of a poorly-coded client.

From the user's point of view, a server that has exited normally has not errored, so to my mind it shouldn't display an error there.  This change was for minimal impact, to avoid rewriting the error code to allow suppressing showing errors to the user in certain cases.